From ec9dd45011b7b3750c0a0955a5f43f214361ed9b Mon Sep 17 00:00:00 2001
From: dylandoesprogramming664 <dylan20xx@outlook.com>
Date: Thu, 15 Jan 2026 12:59:24 -0500
Subject: [PATCH 1/6] DARK Bite, STEEL Magnemite, BANK Opt, Perm Cut

Repo Changes:
- Patched `Makefile` for modern RGBDS
- Updated `rgbdscheck.asm` for modern RGBDS

Gameplay Changes:
- Cut should be permanent thanks to a `wCutTrees` field in `ram/wram.asm`
- Added DARK and STEEL types to RBY Remix; Both are PHYSICAL types

Move Changes (Base/Legacy):
- KARATE_CHOP has 100 accuracy (+5/+5)
- Elemental punches have 75 base power (0/+5)
- FLY has 90 base power (+20/+20)
- HORN_ATTACK is BUG type
- TACKLE is now 50 base power like in Gen V (+15/+15)
- WRAP now has 90 accuracy (+5/+5)
- THRASH now has 120 base power (+30/+30) and 10 PP (-10/-10)
- BITE is now DARK type
- LOW_KICK is now 60 base power (+10/+10)
- PETAL_DANCE now has 120 base power (+30/+30)
- FIRE_SPIN has 35 base power (+20/+20)
- NIGHT_SHADE is now SHADOW_SNEAK, which is a priority move
- LICK is now 60 base power (+40/+20)
- SWIFT can no longer hit through DIG or FLY

Pokemon Changes:
- MAGNEMITE and MAGNETON are now ELECTRIC/STEEL type

Miscellaneous Changes:
- Turned `CalcStats` from `home/move_mon.asm` to
  `engine/battle/calc_stats.asm`

---
 Makefile                               |  14 +-
 constants/move_animation_constants.asm |   2 +-
 constants/move_constants.asm           |   2 +-
 constants/move_effect_constants.asm    |   2 +-
 constants/type_constants.asm           |   4 +-
 data/battle/priority_moves.asm         |   3 +-
 data/moves/animations.asm              |   4 +-
 data/moves/moves.asm                   |  38 ++---
 data/moves/names.asm                   |   2 +-
 data/moves/sfx.asm                     |   2 +-
 data/pokemon/base_stats/magnemite.asm  |   2 +-
 data/pokemon/base_stats/magneton.asm   |   2 +-
 data/pokemon/evos_moves.asm            |  10 +-
 data/predef_pointers.asm               |   1 +
 data/types/names.asm                   |   4 +
 data/types/type_matchups.asm           |  65 ++++++---
 engine/battle/core.asm                 |  93 ++++++------
 engine/overworld/cut.asm               |  15 +-
 engine/overworld/remove_cut_trees.asm  | 161 ++++++++++++++++++++
 engine/overworld/update_map.asm        |  52 ++++---
 engine/pokemon/calc_stats.asm          | 194 ++++++++++++++++++++++++
 home/move_mon.asm                      | 195 +------------------------
 home/overworld.asm                     |   2 +
 main.asm                               |   2 +
 ram/wram.asm                           |   9 +-
 rgbdscheck.asm                         |   8 +-
 26 files changed, 553 insertions(+), 335 deletions(-)
 create mode 100644 engine/overworld/remove_cut_trees.asm
 create mode 100644 engine/pokemon/calc_stats.asm

diff --git a/Makefile b/Makefile
index 5eb584b..6cb1fdb 100644
--- a/Makefile
+++ b/Makefile
@@ -81,7 +81,7 @@ tools:
 	$(MAKE) -C tools/
 
 
-RGBASMFLAGS = -hL -Q8 -P includes.asm -Weverything -Wnumeric-string=2 -Wtruncation=1
+RGBASMFLAGS = -Q8 -P includes.asm -Weverything -Wtruncation=1
 # Create a sym/map for debug purposes if `make` run with `DEBUG=1`
 ifeq ($(DEBUG),1)
 RGBASMFLAGS += -E
@@ -90,7 +90,7 @@ endif
 $(pokeyellow_debug_obj): RGBASMFLAGS += -D _DEBUG
 $(pokeyellow_vc_obj):    RGBASMFLAGS += -D _YELLOW_VC
 
-%.patch: vc/%.constants.sym %_vc.gbc %.gbc vc/%.patch.template
+%.patch: %_vc.gbc %.gbc vc/%.patch.template
 	tools/make_patch $*_vc.sym $^ $@
 
 rgbdscheck.o: rgbdscheck.asm
@@ -116,10 +116,6 @@ $(foreach obj, $(pokeyellow_obj), $(eval $(call DEP,$(obj),$(obj:.o=.asm))))
 $(foreach obj, $(pokeyellow_debug_obj), $(eval $(call DEP,$(obj),$(obj:_debug.o=.asm))))
 $(foreach obj, $(pokeyellow_vc_obj), $(eval $(call DEP,$(obj),$(obj:_vc.o=.asm))))
 
-# Dependencies for VC files that need to run scan_includes
-%.constants.sym: %.constants.asm $(shell tools/scan_includes %.constants.asm) $(preinclude_deps) | rgbdscheck.o
-	$(RGBASM) $(RGBASMFLAGS) $< > $@
-
 endif
 
 
@@ -130,7 +126,7 @@ pokeyellow_pad       = 0x00
 pokeyellow_debug_pad = 0xff
 pokeyellow_vc_pad    = 0x00
 
-opts = -cjsv -k 01 -l 0x33 -m 0x1b -p 0 -r 03 -t "POKEMON YELLOW"
+opts = -cjsv -k 01 -l 0x33 -m MBC5+RAM+BATTERY -p 0 -r 03 -t "POKEMON YELLOW"
 
 %.gbc: $$(%_obj) layout.link
 	$(RGBLINK) -p $($*_pad) -w -m $*.map -n $*.sym -l layout.link -o $@ $(filter %.o,$^)
@@ -169,9 +165,9 @@ gfx/surfing_pikachu/surfing_pikachu_3.2bpp: tools/gfx += --trim-whitespace
 		tools/gfx $(tools/gfx) -o $@ $@)
 
 %.1bpp: %.png
-	$(RGBGFX) $(rgbgfx) -d1 -o $@ $<
+	$(RGBGFX) $(rgbgfx) --depth 1 -o $@ $<
 	$(if $(tools/gfx),\
-		tools/gfx $(tools/gfx) -d1 -o $@ $@)
+		tools/gfx $(tools/gfx) --depth 1 -o $@ $@)
 
 %.pic: %.2bpp
 	tools/pkmncompress $< $@
diff --git a/constants/move_animation_constants.asm b/constants/move_animation_constants.asm
index 2057541..a9bf9a7 100644
--- a/constants/move_animation_constants.asm
+++ b/constants/move_animation_constants.asm
@@ -3,7 +3,7 @@
 	const_def $C0
 DEF FIRST_SE_ID EQU const_value
 	const_skip $18
-	const SE_WAVY_SCREEN               ; $D8 used in Psywave/Night Shade/Psychic etc.
+	const SE_WAVY_SCREEN               ; $D8 used in Psywave/Shadow Sneak/Psychic etc.
 	const SE_SUBSTITUTE_MON            ; $D9 used in Substitute (turns the pokemon into a mini sprite)
 	const SE_SHAKE_BACK_AND_FORTH      ; $DA used in Double Team
 	const SE_SLIDE_ENEMY_MON_OFF       ; $DB used in Whirlwind
diff --git a/constants/move_constants.asm b/constants/move_constants.asm
index 3376c1a..8dfd718 100644
--- a/constants/move_constants.asm
+++ b/constants/move_constants.asm
@@ -106,7 +106,7 @@
 	const QUICK_ATTACK ; 62
 	const RAGE         ; 63
 	const TELEPORT     ; 64
-	const NIGHT_SHADE  ; 65
+	const SHADOW_SNEAK ; 65
 	const MIMIC        ; 66
 	const SCREECH      ; 67
 	const DOUBLE_TEAM  ; 68
diff --git a/constants/move_effect_constants.asm b/constants/move_effect_constants.asm
index 0727f41..27f486d 100644
--- a/constants/move_effect_constants.asm
+++ b/constants/move_effect_constants.asm
@@ -45,7 +45,7 @@
 	const OHKO_EFFECT                ; $26 moves like Horn Drill
 	const CHARGE_EFFECT              ; $27 moves like Solar Beam
 	const SUPER_FANG_EFFECT          ; $28
-	const SPECIAL_DAMAGE_EFFECT      ; $29 Seismic Toss, Night Shade, Sonic Boom, Dragon Rage, Psywave
+	const SPECIAL_DAMAGE_EFFECT      ; $29 Seismic Toss, Sonic Boom, Dragon Rage, Psywave
 	const TRAPPING_EFFECT            ; $2A moves like Wrap
 	const FLY_EFFECT                 ; $2B
 	const ATTACK_TWICE_EFFECT        ; $2C
diff --git a/constants/type_constants.asm b/constants/type_constants.asm
index 82ee6b4..62f80e4 100644
--- a/constants/type_constants.asm
+++ b/constants/type_constants.asm
@@ -10,9 +10,11 @@ DEF PHYSICAL EQU const_value
 	const ROCK         ; $05
 	const BIRD         ; $06
 	const BUG          ; $07
+	const STEEL        ; $08
+	const DARK         ; $09
 
 DEF SPECIAL EQU const_value
-	const GHOST        ; $08
+	const GHOST        ; $0A
 
 DEF UNUSED_TYPES EQU const_value
 	const_next 20
diff --git a/data/battle/priority_moves.asm b/data/battle/priority_moves.asm
index 6a3ddde..623384a 100644
--- a/data/battle/priority_moves.asm
+++ b/data/battle/priority_moves.asm
@@ -4,5 +4,6 @@
 PriorityMovesList:
     db COUNTER,        1
     db QUICK_ATTACK,   8
+    db SHADOW_SNEAK,   8
     db TRANSFORM,      8
-    db -1 ; end
\ No newline at end of file
+    db -1 ; end
diff --git a/data/moves/animations.asm b/data/moves/animations.asm
index a7e1f0b..a3a9621 100644
--- a/data/moves/animations.asm
+++ b/data/moves/animations.asm
@@ -100,7 +100,7 @@ AttackAnimationPointers:
 	dw QuickAttackAnim
 	dw RageAnim
 	dw TeleportAnim
-	dw NightShadeAnim
+	dw ShadowSneakAnim
 	dw MimicAnim
 	dw ScreechAnim
 	dw DoubleTeamAnim
@@ -764,7 +764,7 @@ TeleportAnim:
 	battle_anim NO_MOVE, SE_SHOOT_BALLS_UPWARD
 	db -1 ; end
 
-NightShadeAnim:
+ShadowSneakAnim:
 	battle_anim CONFUSION, SE_FLASH_SCREEN_LONG
 	battle_anim NO_MOVE, SE_WAVY_SCREEN
 	db -1 ; end
diff --git a/data/moves/moves.asm b/data/moves/moves.asm
index 07eb199..0e4d1bc 100644
--- a/data/moves/moves.asm
+++ b/data/moves/moves.asm
@@ -12,14 +12,14 @@ Moves:
 ; Characteristics of each move.
 	table_width MOVE_LENGTH, Moves
 	move POUND,        NO_ADDITIONAL_EFFECT,        40, NORMAL,       100, 35
-	move KARATE_CHOP,  NO_ADDITIONAL_EFFECT,        50, FIGHTING,     95, 25
+	move KARATE_CHOP,  NO_ADDITIONAL_EFFECT,        50, FIGHTING,     100, 25
 	move DOUBLESLAP,   TWO_TO_FIVE_ATTACKS_EFFECT,  20, NORMAL,       100, 35
 	move COMET_PUNCH,  TWO_TO_FIVE_ATTACKS_EFFECT,  25, NORMAL,       100, 25
 	move MEGA_PUNCH,   NO_ADDITIONAL_EFFECT,        80, NORMAL,        85, 20
 	move PAY_DAY,      PAY_DAY_EFFECT,              60, NORMAL,       100, 20
-	move FIRE_PUNCH,   BURN_SIDE_EFFECT2,           70, FIRE,         100, 15
-	move ICE_PUNCH,    FREEZE_SIDE_EFFECT,          70, ICE,          100, 15
-	move THUNDERPUNCH, PARALYZE_SIDE_EFFECT1,       70, ELECTRIC,     100, 15
+	move FIRE_PUNCH,   BURN_SIDE_EFFECT2,           75, FIRE,         100, 15
+	move ICE_PUNCH,    FREEZE_SIDE_EFFECT,          75, ICE,          100, 15
+	move THUNDERPUNCH, PARALYZE_SIDE_EFFECT1,       75, ELECTRIC,     100, 15
 	move SCRATCH,      NO_ADDITIONAL_EFFECT,        40, NORMAL,       100, 35
 	move VICEGRIP,     NO_ADDITIONAL_EFFECT,        55, NORMAL,       100, 30
 	move GUILLOTINE,   OHKO_EFFECT,                  1, NORMAL,        30,  5
@@ -29,9 +29,9 @@ Moves:
 	move GUST,         NO_ADDITIONAL_EFFECT,        40, FLYING,       100, 35
 	move WING_ATTACK,  NO_ADDITIONAL_EFFECT,        60, FLYING,       100, 35
 	move WHIRLWIND,    SWITCH_AND_TELEPORT_EFFECT,   0, BIRD,          85, 20
-	move FLY,          FLY_EFFECT,                  70, FLYING,       100, 15
+	move FLY,          FLY_EFFECT,                  90, FLYING,       100, 15
 	move BIND,         TRAPPING_EFFECT,             15, NORMAL,        85, 20
-	move SLAM,         FLINCH_SIDE_EFFECT1,        80,  DRAGON,       100, 20
+	move SLAM,         FLINCH_SIDE_EFFECT1,         80, DRAGON,       100, 20
 	move VINE_WHIP,    NO_ADDITIONAL_EFFECT,        40, GRASS,        100, 25
 	move STOMP,        FLINCH_SIDE_EFFECT2,         65, NORMAL,       100, 20
 	move DOUBLE_KICK,  ATTACK_TWICE_EFFECT,         30, FIGHTING,     100, 30
@@ -40,21 +40,21 @@ Moves:
 	move ROLLING_KICK, FLINCH_SIDE_EFFECT2,         70, FIGHTING,     100, 15
 	move SAND_ATTACK,  ACCURACY_DOWN1_EFFECT,        0, BIRD,         100, 15
 	move HEADBUTT,     FLINCH_SIDE_EFFECT2,         70, NORMAL,       100, 15
-	move HORN_ATTACK,  NO_ADDITIONAL_EFFECT,        65, NORMAL,       100, 25
+	move HORN_ATTACK,  NO_ADDITIONAL_EFFECT,        65, BUG,          100, 25
 	move FURY_ATTACK,  TWO_TO_FIVE_ATTACKS_EFFECT,  18, NORMAL,       100, 35
 	move HORN_DRILL,   OHKO_EFFECT,                  1, NORMAL,        30,  5
-	move TACKLE,       NO_ADDITIONAL_EFFECT,        35, NORMAL,       100, 35
+	move TACKLE,       NO_ADDITIONAL_EFFECT,        50, NORMAL,       100, 35
 	move BODY_SLAM,    PARALYZE_SIDE_EFFECT2,       85, NORMAL,       100, 15
-	move WRAP,         TRAPPING_EFFECT,             15, NORMAL,        85, 20
+	move WRAP,         TRAPPING_EFFECT,             15, NORMAL,        90, 20
 	move TAKE_DOWN,    RECOIL_EFFECT,               95, NORMAL,       100, 20
-	move THRASH,       THRASH_PETAL_DANCE_EFFECT,   90, NORMAL,       100, 20
+	move THRASH,       THRASH_PETAL_DANCE_EFFECT,  120, NORMAL,       100, 10
 	move DOUBLE_EDGE,  RECOIL_EFFECT,              120, NORMAL,       100, 15
 	move TAIL_WHIP,    DEFENSE_DOWN1_EFFECT,         0, BIRD,         100, 30
 	move POISON_STING, POISON_SIDE_EFFECT1,         35, POISON,       100, 35
 	move TWINEEDLE,    TWINEEDLE_EFFECT,            40, BUG,          100, 20
 	move PIN_MISSILE,  TWO_TO_FIVE_ATTACKS_EFFECT,  20, BUG,          100, 30
 	move LEER,         DEFENSE_DOWN1_EFFECT,         0, BIRD,         100, 30
-	move BITE,         FLINCH_SIDE_EFFECT1,         60, NORMAL,       100, 25
+	move BITE,         FLINCH_SIDE_EFFECT1,         60, DARK,         100, 25
 	move GROWL,        ATTACK_DOWN1_EFFECT,          0, BIRD,         100, 40
 	move ROAR,         SWITCH_AND_TELEPORT_EFFECT,   0, BIRD,         100, 20
 	move SING,         SLEEP_EFFECT,                 0, BIRD,          55, 15
@@ -77,7 +77,7 @@ Moves:
 	move PECK,         NO_ADDITIONAL_EFFECT,        35, FLYING,       100, 35
 	move DRILL_PECK,   NO_ADDITIONAL_EFFECT,        80, FLYING,       100, 20
 	move SUBMISSION,   RECOIL_EFFECT,               80, FIGHTING,     100, 25
-	move LOW_KICK,     FLINCH_SIDE_EFFECT2,         50, FIGHTING,     100, 20
+	move LOW_KICK,     FLINCH_SIDE_EFFECT2,         60, FIGHTING,     100, 20
 	move COUNTER,      NO_ADDITIONAL_EFFECT,         1, FIGHTING,     100, 20
 	move SEISMIC_TOSS, SPECIAL_DAMAGE_EFFECT,        1, FIGHTING,     100, 20
 	move STRENGTH,     NO_ADDITIONAL_EFFECT,        80, NORMAL,       100, 15
@@ -90,15 +90,15 @@ Moves:
 	move POISONPOWDER, POISON_EFFECT,                0, POISON,        90, 35
 	move STUN_SPORE,   PARALYZE_EFFECT,              0, GRASS,         75, 30
 	move SLEEP_POWDER, SLEEP_EFFECT,                 0, GRASS,         75, 15
-	move PETAL_DANCE,  THRASH_PETAL_DANCE_EFFECT,   90, GRASS,        100, 20
+	move PETAL_DANCE,  THRASH_PETAL_DANCE_EFFECT,  120, GRASS,        100, 20
 	move STRING_SHOT,  SPEED_DOWN1_EFFECT,           0, BUG,           95, 40
 	move DRAGON_RAGE,  SPECIAL_DAMAGE_EFFECT,        1, DRAGON,       100, 20
-	move FIRE_SPIN,    TRAPPING_EFFECT,             15, FIRE,          85, 15
+	move FIRE_SPIN,    TRAPPING_EFFECT,             35, FIRE,          85, 15
 	move THUNDERSHOCK, PARALYZE_SIDE_EFFECT1,       40, ELECTRIC,     100, 30
 	move THUNDERBOLT,  PARALYZE_SIDE_EFFECT1,       95, ELECTRIC,     100, 15
 	move THUNDER_WAVE, PARALYZE_EFFECT,              0, ELECTRIC,     100, 20
-	move THUNDER,      PARALYZE_SIDE_EFFECT1,      120, ELECTRIC,      85, 5
-	move ROCK_THROW,   NO_ADDITIONAL_EFFECT,        50, ROCK,         95, 25
+	move THUNDER,      PARALYZE_SIDE_EFFECT1,      120, ELECTRIC,      85,  5
+	move ROCK_THROW,   NO_ADDITIONAL_EFFECT,        50, ROCK,          95, 25
 	move EARTHQUAKE,   NO_ADDITIONAL_EFFECT,       100, GROUND,       100, 10
 	move FISSURE,      OHKO_EFFECT,                  1, GROUND,        30,  5
 	move DIG,          CHARGE_EFFECT,               70, GROUND,       100, 20
@@ -111,7 +111,7 @@ Moves:
 	move QUICK_ATTACK, NO_ADDITIONAL_EFFECT,        40, NORMAL,       100, 30
 	move RAGE,         RAGE_EFFECT,                 60, NORMAL,       100, 20
 	move TELEPORT,     SWITCH_AND_TELEPORT_EFFECT,   0, PSYCHIC_TYPE, 100, 20
-	move NIGHT_SHADE,  NO_ADDITIONAL_EFFECT,        60, GHOST,        100, 15
+	move SHADOW_SNEAK, NO_ADDITIONAL_EFFECT,        40, GHOST,        100, 30
 	move MIMIC,        MIMIC_EFFECT,                 0, BIRD,         100, 10
 	move SCREECH,      DEFENSE_DOWN2_EFFECT,         0, BIRD,          85, 40
 	move DOUBLE_TEAM,  EVASION_UP1_EFFECT,           0, BIRD,         100, 15
@@ -122,7 +122,7 @@ Moves:
 	move CONFUSE_RAY,  CONFUSION_EFFECT,             0, GHOST,        100, 10
 	move WITHDRAW,     DEFENSE_UP1_EFFECT,           0, BIRD,         100, 40
 	move DEFENSE_CURL, DEFENSE_UP1_EFFECT,           0, BIRD,         100, 40
-	move BARRIER,      DEFENSE_UP2_EFFECT,           0, BIRD,	        100, 30
+	move BARRIER,      DEFENSE_UP2_EFFECT,           0, BIRD,	  100, 30
 	move LIGHT_SCREEN, LIGHT_SCREEN_EFFECT,          0, BIRD,         100, 30
 	move HAZE,         HAZE_EFFECT,                  0, BIRD,         100, 30
 	move REFLECT,      REFLECT_EFFECT,               0, BIRD,         100, 20
@@ -132,7 +132,7 @@ Moves:
 	move MIRROR_MOVE,  MIRROR_MOVE_EFFECT,           0, FLYING,       100, 20
 	move SELFDESTRUCT, EXPLODE_EFFECT,             200, NORMAL,       100,  5
 	move EGG_BOMB,     NO_ADDITIONAL_EFFECT,       100, GRASS,        100, 15
-	move LICK,         PARALYZE_SIDE_EFFECT2,       40, GHOST,        100, 30
+	move LICK,         PARALYZE_SIDE_EFFECT2,       60, GHOST,        100, 30
 	move SMOG,         POISON_SIDE_EFFECT2,         40, POISON,        80, 20
 	move SLUDGE,       POISON_SIDE_EFFECT2,         90, POISON,       100, 20
 	move BONE_CLUB,    FLINCH_SIDE_EFFECT1,         65, GROUND,       100, 20
diff --git a/data/moves/names.asm b/data/moves/names.asm
index e959161..28c7a60 100644
--- a/data/moves/names.asm
+++ b/data/moves/names.asm
@@ -100,7 +100,7 @@ MoveNames::
 	li "QUICK ATTACK"
 	li "RAGE"
 	li "TELEPORT"
-	li "NIGHT SHADE"
+	li "SHADOW SNEAK"
 	li "MIMIC"
 	li "SCREECH"
 	li "DOUBLE TEAM"
diff --git a/data/moves/sfx.asm b/data/moves/sfx.asm
index c3f01f4..f328a7a 100644
--- a/data/moves/sfx.asm
+++ b/data/moves/sfx.asm
@@ -101,7 +101,7 @@ MoveSoundTable:
 	db SFX_BATTLE_25,          $00, $10 ; QUICK_ATTACK
 	db SFX_BATTLE_26,          $f0, $20 ; RAGE
 	db SFX_BATTLE_33,          $f0, $c0 ; TELEPORT
-	db SFX_NOT_VERY_EFFECTIVE, $f0, $e0 ; NIGHT_SHADE
+	db SFX_NOT_VERY_EFFECTIVE, $f0, $e0 ; SHADOW_SNEAK
 	db SFX_BATTLE_09,          $f0, $40 ; MIMIC
 	db SFX_BATTLE_31,          $00, $80 ; SCREECH
 	db SFX_BATTLE_33,          $80, $40 ; DOUBLE_TEAM
diff --git a/data/pokemon/base_stats/magnemite.asm b/data/pokemon/base_stats/magnemite.asm
index 6242d69..ceb3ddf 100644
--- a/data/pokemon/base_stats/magnemite.asm
+++ b/data/pokemon/base_stats/magnemite.asm
@@ -3,7 +3,7 @@
 	db  25,  35,  70,  45,  95
 	;   hp  atk  def  spd  spc
 
-	db ELECTRIC, ELECTRIC ; type
+	db ELECTRIC, STEEL ; type
 	db 190 ; catch rate
 	db 89 ; base exp
 
diff --git a/data/pokemon/base_stats/magneton.asm b/data/pokemon/base_stats/magneton.asm
index 2abf05d..bd4eff3 100644
--- a/data/pokemon/base_stats/magneton.asm
+++ b/data/pokemon/base_stats/magneton.asm
@@ -3,7 +3,7 @@
 	db  50,  60,  95,  70, 120
 	;   hp  atk  def  spd  spc
 
-	db ELECTRIC, ELECTRIC ; type
+	db ELECTRIC, STEEL ; type
 	db 60 ; catch rate
 	db 161 ; base exp
 
diff --git a/data/pokemon/evos_moves.asm b/data/pokemon/evos_moves.asm
index 9eb2430..b74dfaf 100644
--- a/data/pokemon/evos_moves.asm
+++ b/data/pokemon/evos_moves.asm
@@ -402,7 +402,7 @@ GengarEvosMoves:
 ; Learnset
 	db 10, SMOG
 	db 15, PSYWAVE
-	db 36, NIGHT_SHADE
+	db 36, SHADOW_SNEAK
 	db 55, HYPNOSIS
 	db 55, DREAM_EATER
 	db 0
@@ -558,7 +558,7 @@ GastlyEvosMoves:
 	db 10, SMOG
 	db 15, PSYWAVE
 	db 23, POISON_GAS
-	db 36, NIGHT_SHADE
+	db 36, SHADOW_SNEAK
 	db 55, HYPNOSIS
 	db 55, DREAM_EATER
 	db 0
@@ -1243,7 +1243,7 @@ VulpixEvosMoves:
 	db 16, CONFUSE_RAY
 	db 25, REFLECT
 	db 32, FLAMETHROWER
-	db 37, NIGHT_SHADE
+	db 37, SHADOW_SNEAK
 	db 42, FIRE_SPIN
 	db 0
 
@@ -1256,7 +1256,7 @@ NinetalesEvosMoves:
 	db 16, CONFUSE_RAY
 	db 25, REFLECT
 	db 32, FLAMETHROWER
-	db 37, NIGHT_SHADE
+	db 37, SHADOW_SNEAK
 	db 42, FIRE_SPIN
 	db 0
 
@@ -2039,7 +2039,7 @@ HaunterEvosMoves:
 ; Learnset
 	db 10, SMOG
 	db 15, PSYWAVE
-	db 36, NIGHT_SHADE
+	db 36, SHADOW_SNEAK
 	db 55, HYPNOSIS
 	db 55, DREAM_EATER
 	db 0
diff --git a/data/predef_pointers.asm b/data/predef_pointers.asm
index eba3a0e..f28c590 100644
--- a/data/predef_pointers.asm
+++ b/data/predef_pointers.asm
@@ -110,3 +110,4 @@ PredefPointers::
 	add_predef DisplayElevatorFloorMenu
 	add_predef OaksAideScript
 	add_predef TryFieldMove
+	add_predef FindTileBlock
diff --git a/data/types/names.asm b/data/types/names.asm
index ac93d57..d5bc8ef 100644
--- a/data/types/names.asm
+++ b/data/types/names.asm
@@ -9,6 +9,8 @@ TypeNames:
 	dw .Rock
 	dw .Normal
 	dw .Bug
+	dw .Steel
+	dw .Dark
 	dw .Ghost
 
 REPT UNUSED_TYPES_END - UNUSED_TYPES
@@ -37,7 +39,9 @@ ENDR
 .Ice:      db "ICE@"
 .Ground:   db "GROUND@"
 .Rock:     db "ROCK@"
+.Steel:    db "STEEL@"
 .Bird:     db "BIRD@"
 .Bug:      db "BUG@"
 .Ghost:    db "GHOST@"
+.Dark:     db "DARK@"
 .Dragon:   db "DRAGON@"
diff --git a/data/types/type_matchups.asm b/data/types/type_matchups.asm
index d776686..c3d6aa8 100644
--- a/data/types/type_matchups.asm
+++ b/data/types/type_matchups.asm
@@ -1,39 +1,45 @@
 TypeEffects:
 	;  attacker,     defender,     *=
-	db WATER,        FIRE,         SUPER_EFFECTIVE
+	db NORMAL,       ROCK,         NOT_VERY_EFFECTIVE
+	db NORMAL,       GHOST,        NO_EFFECT
+	db NORMAL,       STEEL,        NOT_VERY_EFFECTIVE
+	db FIRE,         WATER,        NOT_VERY_EFFECTIVE
+	db FIRE,         FIRE,         NOT_VERY_EFFECTIVE
 	db FIRE,         GRASS,        SUPER_EFFECTIVE
+	db FIRE,         ROCK,         NOT_VERY_EFFECTIVE
+	db FIRE,         BUG,          SUPER_EFFECTIVE
 	db FIRE,         ICE,          SUPER_EFFECTIVE
-	db GRASS,        WATER,        SUPER_EFFECTIVE
-	db ELECTRIC,     WATER,        SUPER_EFFECTIVE
-	db WATER,        ROCK,         SUPER_EFFECTIVE
-	db GROUND,       FLYING,       NO_EFFECT
+	db FIRE,         DRAGON,       NOT_VERY_EFFECTIVE
+	db FIRE,         STEEL,        SUPER_EFFECTIVE
 	db WATER,        WATER,        NOT_VERY_EFFECTIVE
-	db FIRE,         FIRE,         NOT_VERY_EFFECTIVE
-	db ELECTRIC,     ELECTRIC,     NOT_VERY_EFFECTIVE
-	db ICE,          ICE,          NOT_VERY_EFFECTIVE
-	db GRASS,        GRASS,        NOT_VERY_EFFECTIVE
-	db PSYCHIC_TYPE, PSYCHIC_TYPE, NOT_VERY_EFFECTIVE
-	db FIRE,         WATER,        NOT_VERY_EFFECTIVE
-	db GRASS,        FIRE,         NOT_VERY_EFFECTIVE
+	db WATER,        FIRE,         SUPER_EFFECTIVE
+	db WATER,        ROCK,         SUPER_EFFECTIVE
 	db WATER,        GRASS,        NOT_VERY_EFFECTIVE
-	db ELECTRIC,     GRASS,        NOT_VERY_EFFECTIVE
-	db NORMAL,       ROCK,         NOT_VERY_EFFECTIVE
-	db NORMAL,       GHOST,        NO_EFFECT
-	db GHOST,        GHOST,        SUPER_EFFECTIVE
-	db FIRE,         BUG,          SUPER_EFFECTIVE
-	db FIRE,         ROCK,         NOT_VERY_EFFECTIVE
 	db WATER,        GROUND,       SUPER_EFFECTIVE
+	db WATER,        DRAGON,       NOT_VERY_EFFECTIVE
+	db ELECTRIC,     WATER,        SUPER_EFFECTIVE
+	db ELECTRIC,     DRAGON,       NOT_VERY_EFFECTIVE
+	db ELECTRIC,     ELECTRIC,     NOT_VERY_EFFECTIVE
 	db ELECTRIC,     GROUND,       NO_EFFECT
 	db ELECTRIC,     FLYING,       SUPER_EFFECTIVE
+	db ELECTRIC,     GRASS,        NOT_VERY_EFFECTIVE
+	db GRASS,        WATER,        SUPER_EFFECTIVE
+	db GRASS,        DRAGON,       NOT_VERY_EFFECTIVE
+	db GRASS,        GRASS,        NOT_VERY_EFFECTIVE
+	db GRASS,        FIRE,         NOT_VERY_EFFECTIVE
 	db GRASS,        GROUND,       SUPER_EFFECTIVE
 	db GRASS,        BUG,          NOT_VERY_EFFECTIVE
 	db GRASS,        POISON,       NOT_VERY_EFFECTIVE
 	db GRASS,        ROCK,         SUPER_EFFECTIVE
 	db GRASS,        FLYING,       NOT_VERY_EFFECTIVE
+	db GRASS,        STEEL,        NOT_VERY_EFFECTIVE
+	db ICE,          ICE,          NOT_VERY_EFFECTIVE
+	db ICE,          DRAGON,       SUPER_EFFECTIVE
 	db ICE,          WATER,        NOT_VERY_EFFECTIVE
 	db ICE,          GRASS,        SUPER_EFFECTIVE
 	db ICE,          GROUND,       SUPER_EFFECTIVE
 	db ICE,          FLYING,       SUPER_EFFECTIVE
+	db ICE,          STEEL,        NOT_VERY_EFFECTIVE
 	db FIGHTING,     NORMAL,       SUPER_EFFECTIVE
 	db FIGHTING,     POISON,       NOT_VERY_EFFECTIVE
 	db FIGHTING,     FLYING,       NOT_VERY_EFFECTIVE
@@ -42,43 +48,54 @@ TypeEffects:
 	db FIGHTING,     ROCK,         SUPER_EFFECTIVE
 	db FIGHTING,     ICE,          SUPER_EFFECTIVE
 	db FIGHTING,     GHOST,        NO_EFFECT
+	db FIGHTING,     STEEL,        SUPER_EFFECTIVE
 	db POISON,       GRASS,        SUPER_EFFECTIVE
 	db POISON,       POISON,       NOT_VERY_EFFECTIVE
 	db POISON,       GROUND,       NOT_VERY_EFFECTIVE
 	db POISON,       BUG,          SUPER_EFFECTIVE
 	db POISON,       ROCK,         NOT_VERY_EFFECTIVE
 	db POISON,       GHOST,        NOT_VERY_EFFECTIVE
+	db POISON,       STEEL,        NO_EFFECT
 	db GROUND,       FIRE,         SUPER_EFFECTIVE
-	db GROUND,       ELECTRIC,     SUPER_EFFECTIVE
 	db GROUND,       GRASS,        NOT_VERY_EFFECTIVE
+	db GROUND,       ELECTRIC,     SUPER_EFFECTIVE
+	db GROUND,       FLYING,       NO_EFFECT
 	db GROUND,       BUG,          NOT_VERY_EFFECTIVE
 	db GROUND,       ROCK,         SUPER_EFFECTIVE
 	db GROUND,       POISON,       SUPER_EFFECTIVE
+	db GROUND,       STEEL,        SUPER_EFFECTIVE
 	db FLYING,       ELECTRIC,     NOT_VERY_EFFECTIVE
 	db FLYING,       FIGHTING,     SUPER_EFFECTIVE
 	db FLYING,       BUG,          SUPER_EFFECTIVE
 	db FLYING,       GRASS,        SUPER_EFFECTIVE
 	db FLYING,       ROCK,         NOT_VERY_EFFECTIVE
+	db FLYING,       STEEL,        NOT_VERY_EFFECTIVE
 	db PSYCHIC_TYPE, FIGHTING,     SUPER_EFFECTIVE
 	db PSYCHIC_TYPE, POISON,       SUPER_EFFECTIVE
+	db PSYCHIC_TYPE, PSYCHIC_TYPE, NOT_VERY_EFFECTIVE
+	db PSYCHIC_TYPE, STEEL,        NOT_VERY_EFFECTIVE
 	db BUG,          FIRE,         NOT_VERY_EFFECTIVE
 	db BUG,          GRASS,        SUPER_EFFECTIVE
 	db BUG,          FIGHTING,     NOT_VERY_EFFECTIVE
 	db BUG,          FLYING,       NOT_VERY_EFFECTIVE
 	db BUG,          PSYCHIC_TYPE, SUPER_EFFECTIVE
 	db BUG,          GHOST,        NOT_VERY_EFFECTIVE
+	db BUG,          STEEL,        NOT_VERY_EFFECTIVE
 	db ROCK,         FIRE,         SUPER_EFFECTIVE
 	db ROCK,         FIGHTING,     NOT_VERY_EFFECTIVE
 	db ROCK,         GROUND,       NOT_VERY_EFFECTIVE
 	db ROCK,         FLYING,       SUPER_EFFECTIVE
 	db ROCK,         BUG,          SUPER_EFFECTIVE
 	db ROCK,         ICE,          SUPER_EFFECTIVE
+	db ROCK,         STEEL,        NOT_VERY_EFFECTIVE
 	db GHOST,        NORMAL,       NO_EFFECT
 	db GHOST,        PSYCHIC_TYPE, SUPER_EFFECTIVE
-	db FIRE,         DRAGON,       NOT_VERY_EFFECTIVE
-	db WATER,        DRAGON,       NOT_VERY_EFFECTIVE
-	db ELECTRIC,     DRAGON,       NOT_VERY_EFFECTIVE
-	db GRASS,        DRAGON,       NOT_VERY_EFFECTIVE
-	db ICE,          DRAGON,       SUPER_EFFECTIVE
+	db GHOST,        GHOST,        SUPER_EFFECTIVE
+	db GHOST,        STEEL,        NOT_VERY_EFFECTIVE
 	db DRAGON,       DRAGON,       SUPER_EFFECTIVE
+        db DRAGON,       STEEL,        NOT_VERY_EFFECTIVE
+	db DARK,         PSYCHIC_TYPE, SUPER_EFFECTIVE
+	db DARK,         GHOST,        SUPER_EFFECTIVE
+	db DARK,         FIGHTING,     NOT_VERY_EFFECTIVE
+        db DARK,         STEEL,        NOT_VERY_EFFECTIVE
 	db -1 ; end
diff --git a/engine/battle/core.asm b/engine/battle/core.asm
index b9ae299..5c33231 100644
--- a/engine/battle/core.asm
+++ b/engine/battle/core.asm
@@ -473,41 +473,41 @@ HandleMovePriority:
 ; the enemy's priority value will be stored in register e.
 ; These values will be compared after the 'ret' instruction is called
 
-        ld a, [wPlayerSelectedMove]
-        ld b, a
-        ld hl, PriorityMovesList
-        ld c, 7           ; no priority is 7
+	ld a, [wPlayerSelectedMove]
+	ld b, a
+	ld hl, PriorityMovesList
+	ld c, 7           ; no priority is 7
 .playerPriorityMoveLoop
-        ld a, [hli]       ; load the move ID from priority list and 
-                          ; increment address to the priority value address
-        cp b              ; compare with move being used
-        jr z, .playerUsingPriorityMove
-        inc a             ; if at end of list: -1 + 1 = 0xFF + 0x01 = 0
-        jr z, .noPlayerPriorityMove
-        inc hl            ; increment address to the next move
-        jr .playerPriorityMoveLoop
+	ld a, [hli]       ; load the move ID from priority list and
+	                  ; increment address to the priority value address
+	cp b              ; compare with move being used
+	jr z, .playerUsingPriorityMove
+	inc a             ; if at end of list: -1 + 1 = 0xFF + 0x01 = 0
+	jr z, .noPlayerPriorityMove
+	inc hl            ; increment address to the next move
+	jr .playerPriorityMoveLoop
 .playerUsingPriorityMove
-        ld c, [hl]        ; get new priority value 
+	ld c, [hl]        ; get new priority value
 .noPlayerPriorityMove
 
 ; Now check enemy priority 
-        ld a, [wEnemySelectedMove]
-        ld d, a
-        ld hl, PriorityMovesList
-        ld e, 7           ; no priority is 7
+	ld a, [wEnemySelectedMove]
+	ld d, a
+	ld hl, PriorityMovesList
+	ld e, 7           ; no priority is 7
 .enemyPriorityMoveLoop
-        ld a, [hli]       ; load the move ID from priority list and 
-                          ; increment address to the priority value address
-        cp d              ; compare with move being used
-        jr z, .enemyUsingPriorityMove
-        inc a             ; if at end of list: -1 + 1 = 0xFF + 0x01 = 0
-        jr z, .noEnemyPriorityMove
-        inc hl            ; increment address to the next move
-        jr .enemyPriorityMoveLoop
+	ld a, [hli]       ; load the move ID from priority list and
+	                  ; increment address to the priority value address
+	cp d              ; compare with move being used
+	jr z, .enemyUsingPriorityMove
+	inc a             ; if at end of list: -1 + 1 = 0xFF + 0x01 = 0
+	jr z, .noEnemyPriorityMove
+	inc hl            ; increment address to the next move
+	jr .enemyPriorityMoveLoop
 .enemyUsingPriorityMove
-        ld e, [hl]        ; get new priority value 
+	ld e, [hl]        ; get new priority value
 .noEnemyPriorityMove
-        ret
+	ret
 
 INCLUDE "data/battle/priority_moves.asm"
 
@@ -4156,14 +4156,14 @@ CheckForDisobedience:
 	cp 5
 	ld a, 50 ; Alakazam's level
 	jr nc, .next
-    cp 4
+	cp 4
 	ld a, 43 ; Venomoth's level
 	jr nc, .next
 	cp 3
 	ld a, 35 ; Vileplume's level
 	jr nc, .next
 	cp 2
-    ld a, 24 ; Bit below Raichu's level
+	ld a, 24 ; Bit below Raichu's level
 	jr nc, .next
 	cp 1
 	ld a, 21 ; Starmie's level
@@ -4979,8 +4979,6 @@ ApplyAttackToEnemyPokemon:
 	ld a, [wPlayerMoveNum]
 	cp SEISMIC_TOSS
 	jr z, .storeDamage
-	cp NIGHT_SHADE
-	jr z, .storeDamage
 	ld b, SONICBOOM_DAMAGE ; 20
 	cp SONICBOOM
 	jr z, .storeDamage
@@ -5102,7 +5100,7 @@ ApplyAttackToPlayerPokemon:
 	ld a, [wEnemyMoveNum]
 	cp SEISMIC_TOSS
 	jr z, .storeDamage
-	cp NIGHT_SHADE
+	cp SHADOW_SNEAK
 	jr z, .storeDamage
 	ld b, SONICBOOM_DAMAGE
 	cp SONICBOOM
@@ -5608,24 +5606,25 @@ MoveHitTest:
 .dreamEaterCheck
 	ld a, [de]
 	cp DREAM_EATER_EFFECT
-	jr nz, .swiftCheck
+	jr nz, .checkForDigOrFlyStatus
 	ld a, [bc]
 	and SLP_MASK
 	jp z, .moveMissed
+.checkForDigOrFlyStatus
+	bit INVULNERABLE, [hl]
+	jp nz, .moveMissed
 .swiftCheck
 	ld a, [de]
 	cp SWIFT_EFFECT
 	ret z ; Swift never misses (this was fixed from the Japanese versions)
 	call CheckTargetSubstitute ; substitute check (note that this overwrites a)
-	jr z, .checkForDigOrFlyStatus
+	jr z, .noSubstitute
 	ld a, [de]
 	cp DRAIN_HP_EFFECT
 	jp z, .moveMissed
 	cp DREAM_EATER_EFFECT
 	jp z, .moveMissed
-.checkForDigOrFlyStatus
-	bit INVULNERABLE, [hl]
-	jp nz, .moveMissed
+.noSubstitute
 	ldh a, [hWhoseTurn]
 	and a
 	jr nz, .enemyTurn
@@ -6491,13 +6490,13 @@ LoadEnemyMonData:
 	ld a, [hl]     ; base exp
 	ld [de], a
 	; Nickname code
-    ld a, [wCurOpponent]
-    cp OPP_CRAIG
-    jr nz, .loadSpeciesName
-    ld hl, .CraigMonsNicks
-    ld a, [wWhichPokemon]
-    call SkipFixedLengthTextEntries
-    jr .copyNick
+	ld a, [wCurOpponent]
+	cp OPP_CRAIG
+	jr nz, .loadSpeciesName
+	ld hl, .CraigMonsNicks
+	ld a, [wWhichPokemon]
+	call SkipFixedLengthTextEntries
+	jr .copyNick
 .loadSpeciesName
 	ld a, [wEnemyMonSpecies2]
 	ld [wd11e], a
@@ -6881,9 +6880,9 @@ ApplyBadgeStatBoosts:
 	rr e
 	srl d
 	rr e
-    ld a, [wBattleMonSpecies] ; Check if the species is Pikachu
-    cp PIKACHU
-    jr nz, .continue ; times by 1.25 if its Pikachu
+	ld a, [wBattleMonSpecies] ; Check if the species is Pikachu
+	cp PIKACHU
+	jr nz, .continue ; times by 1.25 if its Pikachu
 	srl d
 	rr e
 .continue
diff --git a/engine/overworld/cut.asm b/engine/overworld/cut.asm
index 9f5b7b6..54509b6 100644
--- a/engine/overworld/cut.asm
+++ b/engine/overworld/cut.asm
@@ -59,6 +59,7 @@ Cut2::
 	call InitCutAnimOAM
 	ld de, CutTreeBlockSwaps
 	call ReplaceTreeTileBlock
+	farcall SetCutTreeFlags
 	call RedrawMapView
 	farcall AnimCut
 	ld a, $1
@@ -239,6 +240,11 @@ ReplaceTreeTileBlock:
 	pop de
 	ld a, [hl]
 	ld c, a
+	call LoopForTileReplacement
+	ld [hl], a
+	ret
+
+LoopForTileReplacement:
 .loop ; find the matching tile block in the array
 	ld a, [de]
 	inc de
@@ -249,7 +255,14 @@ ReplaceTreeTileBlock:
 	jr nz, .loop
 	dec de
 	ld a, [de] ; replacement tile block from matching array entry
-	ld [hl], a
+	ret
+
+FindTileBlockReplacementCut::
+	ld de, CutTreeBlockSwaps
+	ld a, [wNewTileBlockID]
+	ld c, a
+	call LoopForTileReplacement
+	ld [wNewTileBlockID], a
 	ret
 
 INCLUDE "data/tilesets/cut_tree_blocks.asm"
diff --git a/engine/overworld/remove_cut_trees.asm b/engine/overworld/remove_cut_trees.asm
new file mode 100644
index 0000000..40d8e7f
--- /dev/null
+++ b/engine/overworld/remove_cut_trees.asm
@@ -0,0 +1,161 @@
+CutTreeLocations:
+; first byte = The map the tree is on
+; second byte = The Y coordinate of the block
+; third byte = The X coordinate of the block
+	db VIRIDIAN_CITY, 2, 7
+	db VIRIDIAN_CITY, 11, 4
+	db ROUTE_2, 11, 7
+	db ROUTE_2, 26, 6
+	db ROUTE_2, 30, 6
+	db ROUTE_2, 34, 6
+	db PEWTER_CITY, 2, 13
+	db CERULEAN_CITY, 14, 9
+	db ROUTE_8, 5, 20
+	db ROUTE_8, 6, 14
+	db ROUTE_9, 4, 2
+	db VERMILION_CITY, 9, 7
+	db ROUTE_10, 9, 4
+	db ROUTE_10, 10, 4
+	db ROUTE_10, 11, 4
+	db ROUTE_10, 12, 4
+	db CELADON_CITY, 10, 23
+	db ROUTE_13, 2, 17
+	db ROUTE_14, 16, 5
+	db ROUTE_14, 13, 2
+	db ROUTE_14, 21, 1
+	db ROUTE_16, 4, 17
+	db ROUTE_25, 1, 13
+	db $FF ; list terminator
+
+SetCutTreeFlags::
+	ld a, [wYCoord]
+	sra a
+	ld d, a ; d holds the Y block loc
+	ld a, [wXCoord]
+	sra a
+	ld e, a ; e holds the X block loc
+	ld a, [wSpriteStateData1 + 9] ; player sprite's facing direction
+	and a
+	jr z, .down
+	cp SPRITE_FACING_UP
+	jr z, .up
+	cp SPRITE_FACING_LEFT
+	jr z, .left
+; right
+	ld a, [wXBlockCoord]
+	and a
+	jr z, .findMapLoc
+	inc e
+	jr .findMapLoc
+.down
+	ld a, [wYBlockCoord]
+	and a
+	jr z, .findMapLoc
+	inc d
+	jr .findMapLoc
+.up
+	ld a, [wYBlockCoord]
+	and a
+	jr nz, .findMapLoc
+	dec d
+	jr .findMapLoc
+.left
+	ld a, [wXBlockCoord]
+	and a
+	jr nz, .findMapLoc
+	dec e
+.findMapLoc
+	ld a,[wCurMap]
+	ld b, a
+	ld hl, CutTreeLocations ; d = Y loc ; e = X loc ; b = map loc
+	ld c, 0
+	jr .loopfirst
+.loopinctwo
+	inc hl
+.loopincone
+	inc hl
+	inc c
+.loopfirst ; find the matching tile block in the array
+	ld a, [hl]
+	inc hl
+	cp $ff
+	ret z ; Not in list; return with a cleared carry flag
+	cp b ; Compare map
+	jr nz, .loopinctwo
+	ld a, [hl]  ; hl +1 (Y loc)
+	inc hl
+	cp d
+	jr nz, .loopincone
+	ld a, [hl]  ; hl +2 (X loc)
+	cp e
+	jr nz, .loopincone
+	ld b, 1
+	ld hl, wCutTrees
+	ld a, c
+	cp 8
+	jr c, .setByte
+	; second byte of wCutTrees
+	sub 8
+	inc hl
+	ld c, a
+.setByte
+	predef FlagActionPredef
+	ret
+
+RemoveAlreadyCutTrees::
+	ld hl, CutTreeLocations
+	ld c, 0
+	jr .loopfirst
+.loopinctwo
+	inc hl
+.loopincone
+	inc hl
+	inc c
+.loopfirst ; find the matching tile block in the array
+	ld a,[wCurMap]
+	ld b, a
+	ld a, [hl]
+	inc hl
+	cp $ff
+	ret z ; Not in list; return with a cleared carry flag
+	cp b ; Compare map
+	jr nz, .loopinctwo
+
+	ld d, [hl]  ; hl +1 (Y loc)
+	inc hl
+	ld e, [hl]  ; hl +2 (X loc)
+	push hl
+	ld hl, wCutTrees
+	ld a, c
+	ld [wTempCoins1], a ; temporarily store the current iteration
+.iterByte
+	cp 8
+	jr c, .checkByte
+	sub 8
+	inc hl
+	ld c, a
+	jr .iterByte
+.checkByte
+	ld b, 2
+	predef FlagActionPredef
+	ld a, c
+	and a
+	ld a, [wTempCoins1]
+	ld c, a
+	pop hl
+	jr z, .loopincone
+	ld b, d
+	ld c, e
+	push hl
+	predef FindTileBlock ; hl holds block ID at X,Y coord on the map
+	ld a, [hl]
+	ld [wNewTileBlockID], a
+	push hl
+	farcall FindTileBlockReplacementCut
+	pop hl
+	ld a, [wNewTileBlockID]
+	ld [hl], a
+	ld a, [wTempCoins1]
+	ld c, a
+	pop hl
+	jr .loopincone
diff --git a/engine/overworld/update_map.asm b/engine/overworld/update_map.asm
index dbf28da..0703a67 100644
--- a/engine/overworld/update_map.asm
+++ b/engine/overworld/update_map.asm
@@ -3,28 +3,7 @@
 ; b = Y
 ; c = X
 ReplaceTileBlock:
-	call GetPredefRegisters
-	ld hl, wOverworldMap
-	ld a, [wCurMapWidth]
-	add $6
-	ld e, a
-	ld d, $0
-	add hl, de
-	add hl, de
-	add hl, de
-	ld e, $3
-	add hl, de
-	ld e, a
-	ld a, b
-	and a
-	jr z, .addX
-; add width * Y
-.addWidthYTimesLoop
-	add hl, de
-	dec b
-	jr nz, .addWidthYTimesLoop
-.addX
-	add hl, bc ; add X
+	call FindTileBlock
 	ld a, [wNewTileBlockID]
 	ld [hl], a
 	ld a, [wCurrentTileBlockMapViewPointer]
@@ -124,3 +103,32 @@ CompareHLWithBC:
 	ld a, l
 	sub c
 	ret
+
+; loads the current tile block into in [wNewTileBlockID]
+; b = Y
+; c = X
+; ret = hl = the ID of the currently loaded tile
+FindTileBlock:
+	call GetPredefRegisters
+	ld hl, wOverworldMap
+	ld a, [wCurMapWidth]
+	add $6
+	ld e, a
+	ld d, $0
+	add hl, de
+	add hl, de
+	add hl, de
+	ld e, $3
+	add hl, de
+	ld e, a
+	ld a, b
+	and a
+	jr z, .addX
+; add width * Y
+.addWidthYTimesLoop
+	add hl, de
+	dec b
+	jr nz, .addWidthYTimesLoop
+.addX
+	add hl, bc ; add X
+	ret
diff --git a/engine/pokemon/calc_stats.asm b/engine/pokemon/calc_stats.asm
new file mode 100644
index 0000000..1e3c233
--- /dev/null
+++ b/engine/pokemon/calc_stats.asm
@@ -0,0 +1,194 @@
+_CalcStats::
+	ld c, $0
+.statsLoop
+	inc c
+	call _CalcStat
+	ldh a, [hMultiplicand+1]
+	ld [de], a
+	inc de
+	ldh a, [hMultiplicand+2]
+	ld [de], a
+	inc de
+	ld a, c
+	cp NUM_STATS
+	jr nz, .statsLoop
+	ret
+
+_CalcStat::
+	push hl
+	push de
+	push bc
+	ld a, b
+	ld d, a
+	push hl
+	ld hl, wMonHeader
+	ld b, $0
+	add hl, bc
+	ld a, [hl]          ; read base value of stat
+	ld e, a
+	pop hl
+	push hl
+	sla c
+	ld a, d
+	and a
+	jr z, .statExpDone  ; consider stat exp?
+	add hl, bc          ; skip to corresponding stat exp value
+.statExpLoop            ; calculates ceil(Sqrt(stat exp)) in b
+	xor a
+	ldh [hMultiplicand], a
+	ldh [hMultiplicand+1], a
+	inc b               ; increment current stat exp bonus
+	ld a, b
+	cp $ff
+	jr z, .statExpDone
+	ldh [hMultiplicand+2], a
+	ldh [hMultiplier], a
+	call Multiply
+	ld a, [hld]
+	ld d, a
+	ldh a, [hProduct + 3]
+	sub d
+	ld a, [hli]
+	ld d, a
+	ldh a, [hProduct + 2]
+	sbc d               ; test if (current stat exp bonus)^2 < stat exp
+	jr c, .statExpLoop
+.statExpDone
+	srl c
+	pop hl
+	push bc
+	ld bc, wPartyMon1DVs - (wPartyMon1HPExp - 1) ; also wEnemyMonDVs - wEnemyMonHP
+	add hl, bc
+	pop bc
+	ld a, c
+	cp $2
+	jr z, .getAttackIV
+	cp $3
+	jr z, .getDefenseIV
+	cp $4
+	jr z, .getSpeedIV
+	cp $5
+	jr z, .getSpecialIV
+.getHpIV
+	push bc
+	ld a, [hl]  ; Atk IV
+	swap a
+	and $1
+	sla a
+	sla a
+	sla a
+	ld b, a
+	ld a, [hli] ; Def IV
+	and $1
+	sla a
+	sla a
+	add b
+	ld b, a
+	ld a, [hl] ; Spd IV
+	swap a
+	and $1
+	sla a
+	add b
+	ld b, a
+	ld a, [hl] ; Spc IV
+	and $1
+	add b      ; HP IV: LSB of the other 4 IVs
+	pop bc
+	jr .calcStatFromIV
+.getAttackIV
+	ld a, [hl]
+	swap a
+	and $f
+	jr .calcStatFromIV
+.getDefenseIV
+	ld a, [hl]
+	and $f
+	jr .calcStatFromIV
+.getSpeedIV
+	inc hl
+	ld a, [hl]
+	swap a
+	and $f
+	jr .calcStatFromIV
+.getSpecialIV
+	inc hl
+	ld a, [hl]
+	and $f
+.calcStatFromIV
+	ld d, $0
+	add e
+	ld e, a
+	jr nc, .noCarry
+	inc d                     ; de = Base + IV
+.noCarry
+	sla e
+	rl d                      ; de = (Base + IV) * 2
+	srl b
+	srl b                     ; b = ceil(Sqrt(stat exp)) / 4
+	ld a, b
+	add e
+	jr nc, .noCarry2
+	inc d                     ; de = (Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4
+.noCarry2
+	ldh [hMultiplicand+2], a
+	ld a, d
+	ldh [hMultiplicand+1], a
+	xor a
+	ldh [hMultiplicand], a
+	ld a, [wCurEnemyLVL]
+	ldh [hMultiplier], a
+	call Multiply            ; ((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level
+	ldh a, [hMultiplicand]
+	ldh [hDividend], a
+	ldh a, [hMultiplicand+1]
+	ldh [hDividend+1], a
+	ldh a, [hMultiplicand+2]
+	ldh [hDividend+2], a
+	ld a, $64
+	ldh [hDivisor], a
+	ld a, $3
+	ld b, a
+	call Divide             ; (((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level) / 100
+	ld a, c
+	cp $1
+	ld a, 5 ; + 5 for non-HP stat
+	jr nz, .notHPStat
+	ld a, [wCurEnemyLVL]
+	ld b, a
+	ldh a, [hMultiplicand+2]
+	add b
+	ldh [hMultiplicand+2], a
+	jr nc, .noCarry3
+	ldh a, [hMultiplicand+1]
+	inc a
+	ldh [hMultiplicand+1], a ; HP: (((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level) / 100 + Level
+.noCarry3
+	ld a, 10 ; +10 for HP stat
+.notHPStat
+	ld b, a
+	ldh a, [hMultiplicand+2]
+	add b
+	ldh [hMultiplicand+2], a
+	jr nc, .noCarry4
+	ldh a, [hMultiplicand+1]
+	inc a                    ; non-HP: (((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level) / 100 + 5
+	ldh [hMultiplicand+1], a ; HP: (((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level) / 100 + Level + 10
+.noCarry4
+	ldh a, [hMultiplicand+1] ; check for overflow (>999)
+	cp HIGH(MAX_STAT_VALUE) + 1
+	jr nc, .overflow
+	cp HIGH(MAX_STAT_VALUE)
+	jr c, .noOverflow
+	ldh a, [hMultiplicand+2]
+	cp LOW(MAX_STAT_VALUE) + 1
+	jr c, .noOverflow
+.overflow
+	ld a, HIGH(MAX_STAT_VALUE) ; overflow: cap at 999
+	ldh [hMultiplicand+1], a
+	ld a, LOW(MAX_STAT_VALUE)
+	ldh [hMultiplicand+2], a
+.noOverflow
+	pop bc
+	pop de
+	pop hl
+	ret
diff --git a/home/move_mon.asm b/home/move_mon.asm
index c766fbd..065a3e1 100644
--- a/home/move_mon.asm
+++ b/home/move_mon.asm
@@ -32,203 +32,16 @@ AddPartyMon::
 
 ; calculates all 5 stats of current mon and writes them to [de]
 CalcStats::
-	ld c, $0
-.statsLoop
-	inc c
-	call CalcStat
-	ldh a, [hMultiplicand+1]
-	ld [de], a
-	inc de
-	ldh a, [hMultiplicand+2]
-	ld [de], a
-	inc de
-	ld a, c
-	cp NUM_STATS
-	jr nz, .statsLoop
-	ret
+	homecall _CalcStats
+    ret
 
 ; calculates stat c of current mon
 ; c: stat to calc (HP=1,Atk=2,Def=3,Spd=4,Spc=5)
 ; b: consider stat exp?
 ; hl: base ptr to stat exp values ([hl + 2*c - 1] and [hl + 2*c])
 CalcStat::
-	push hl
-	push de
-	push bc
-	ld a, b
-	ld d, a
-	push hl
-	ld hl, wMonHeader
-	ld b, $0
-	add hl, bc
-	ld a, [hl]          ; read base value of stat
-	ld e, a
-	pop hl
-	push hl
-	sla c
-	ld a, d
-	and a
-	jr z, .statExpDone  ; consider stat exp?
-	add hl, bc          ; skip to corresponding stat exp value
-.statExpLoop            ; calculates ceil(Sqrt(stat exp)) in b
-	xor a
-	ldh [hMultiplicand], a
-	ldh [hMultiplicand+1], a
-	inc b               ; increment current stat exp bonus
-	ld a, b
-	cp $ff
-	jr z, .statExpDone
-	ldh [hMultiplicand+2], a
-	ldh [hMultiplier], a
-	call Multiply
-	ld a, [hld]
-	ld d, a
-	ldh a, [hProduct + 3]
-	sub d
-	ld a, [hli]
-	ld d, a
-	ldh a, [hProduct + 2]
-	sbc d               ; test if (current stat exp bonus)^2 < stat exp
-	jr c, .statExpLoop
-.statExpDone
-	srl c
-	pop hl
-	push bc
-	ld bc, wPartyMon1DVs - (wPartyMon1HPExp - 1) ; also wEnemyMonDVs - wEnemyMonHP
-	add hl, bc
-	pop bc
-	ld a, c
-	cp $2
-	jr z, .getAttackIV
-	cp $3
-	jr z, .getDefenseIV
-	cp $4
-	jr z, .getSpeedIV
-	cp $5
-	jr z, .getSpecialIV
-.getHpIV
-	push bc
-	ld a, [hl]  ; Atk IV
-	swap a
-	and $1
-	sla a
-	sla a
-	sla a
-	ld b, a
-	ld a, [hli] ; Def IV
-	and $1
-	sla a
-	sla a
-	add b
-	ld b, a
-	ld a, [hl] ; Spd IV
-	swap a
-	and $1
-	sla a
-	add b
-	ld b, a
-	ld a, [hl] ; Spc IV
-	and $1
-	add b      ; HP IV: LSB of the other 4 IVs
-	pop bc
-	jr .calcStatFromIV
-.getAttackIV
-	ld a, [hl]
-	swap a
-	and $f
-	jr .calcStatFromIV
-.getDefenseIV
-	ld a, [hl]
-	and $f
-	jr .calcStatFromIV
-.getSpeedIV
-	inc hl
-	ld a, [hl]
-	swap a
-	and $f
-	jr .calcStatFromIV
-.getSpecialIV
-	inc hl
-	ld a, [hl]
-	and $f
-.calcStatFromIV
-	ld d, $0
-	add e
-	ld e, a
-	jr nc, .noCarry
-	inc d                     ; de = Base + IV
-.noCarry
-	sla e
-	rl d                      ; de = (Base + IV) * 2
-	srl b
-	srl b                     ; b = ceil(Sqrt(stat exp)) / 4
-	ld a, b
-	add e
-	jr nc, .noCarry2
-	inc d                     ; de = (Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4
-.noCarry2
-	ldh [hMultiplicand+2], a
-	ld a, d
-	ldh [hMultiplicand+1], a
-	xor a
-	ldh [hMultiplicand], a
-	ld a, [wCurEnemyLVL]
-	ldh [hMultiplier], a
-	call Multiply            ; ((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level
-	ldh a, [hMultiplicand]
-	ldh [hDividend], a
-	ldh a, [hMultiplicand+1]
-	ldh [hDividend+1], a
-	ldh a, [hMultiplicand+2]
-	ldh [hDividend+2], a
-	ld a, $64
-	ldh [hDivisor], a
-	ld a, $3
-	ld b, a
-	call Divide             ; (((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level) / 100
-	ld a, c
-	cp $1
-	ld a, 5 ; + 5 for non-HP stat
-	jr nz, .notHPStat
-	ld a, [wCurEnemyLVL]
-	ld b, a
-	ldh a, [hMultiplicand+2]
-	add b
-	ldh [hMultiplicand+2], a
-	jr nc, .noCarry3
-	ldh a, [hMultiplicand+1]
-	inc a
-	ldh [hMultiplicand+1], a ; HP: (((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level) / 100 + Level
-.noCarry3
-	ld a, 10 ; +10 for HP stat
-.notHPStat
-	ld b, a
-	ldh a, [hMultiplicand+2]
-	add b
-	ldh [hMultiplicand+2], a
-	jr nc, .noCarry4
-	ldh a, [hMultiplicand+1]
-	inc a                    ; non-HP: (((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level) / 100 + 5
-	ldh [hMultiplicand+1], a ; HP: (((Base + IV) * 2 + ceil(Sqrt(stat exp)) / 4) * Level) / 100 + Level + 10
-.noCarry4
-	ldh a, [hMultiplicand+1] ; check for overflow (>999)
-	cp HIGH(MAX_STAT_VALUE) + 1
-	jr nc, .overflow
-	cp HIGH(MAX_STAT_VALUE)
-	jr c, .noOverflow
-	ldh a, [hMultiplicand+2]
-	cp LOW(MAX_STAT_VALUE) + 1
-	jr c, .noOverflow
-.overflow
-	ld a, HIGH(MAX_STAT_VALUE) ; overflow: cap at 999
-	ldh [hMultiplicand+1], a
-	ld a, LOW(MAX_STAT_VALUE)
-	ldh [hMultiplicand+2], a
-.noOverflow
-	pop bc
-	pop de
-	pop hl
-	ret
+	homecall _CalcStat
+    ret
 
 AddEnemyMonToPlayerParty::
 	homecall_sf _AddEnemyMonToPlayerParty
diff --git a/home/overworld.asm b/home/overworld.asm
index 5a6e2f2..a27b010 100644
--- a/home/overworld.asm
+++ b/home/overworld.asm
@@ -708,6 +708,7 @@ CheckMapConnections::
 ; x#SPRITESTATEDATA2_IMAGEBASEOFFSET without loading any tile patterns.
 	call InitMapSprites
 	call LoadTileBlockMap
+	farcall RemoveAlreadyCutTrees
 	jp OverworldLoopLessDelay
 
 .didNotEnterConnectedMap
@@ -2051,6 +2052,7 @@ LoadMapData::
 
 LoadScreenRelatedData::
 	call LoadTileBlockMap
+	farcall RemoveAlreadyCutTrees
 	call LoadTilesetTilePatternData
 	call LoadCurrentMapView
 	ret
diff --git a/main.asm b/main.asm
index a6a0d27..dd347dc 100644
--- a/main.asm
+++ b/main.asm
@@ -305,6 +305,7 @@ INCLUDE "engine/items/itemfinder.asm"
 SECTION "Vending Machine", ROMX
 
 INCLUDE "engine/events/vending_machine.asm"
+INCLUDE "engine/pokemon/calc_stats.asm"
 
 
 SECTION "Itemfinder 2", ROMX
@@ -416,6 +417,7 @@ INCLUDE "engine/battle/get_trainer_name.asm"
 INCLUDE "engine/math/random.asm"
 INCLUDE "engine/predefs.asm"
 INCLUDE "engine/overworld/field_moves.asm"
+INCLUDE "engine/overworld/remove_cut_trees.asm"
 
 
 SECTION "Surfing Minigame", ROMX
diff --git a/ram/wram.asm b/ram/wram.asm
index fca583d..da845f3 100644
--- a/ram/wram.asm
+++ b/ram/wram.asm
@@ -210,7 +210,6 @@ wc5d2:: db
 wSurfingMinigameWaveFunctionNumber:: dw
 wc5d5:: db
 wSurfingMinigamePikachuHP:: dw ; little-endian BCD
-wc5d8:: db ; unused?
 ; number of consecutive tricks
 wSurfingMinigameRadnessMeter:: db
 wSurfingMinigameRadnessScore:: dw ; little-endian BCD
@@ -558,7 +557,13 @@ NEXTU
 wBoostExpByExpAll::
 wUnusedCC5B:: db
 
-	ds 59
+	ds 53
+
+wCutTrees::
+; Check CutTreeLocations for the indexes
+	ds 3
+	; unused
+	ds 3
 
 wNPCMovementDirections2:: ds 10
 ; used in Pallet Town scripted movement
diff --git a/rgbdscheck.asm b/rgbdscheck.asm
index fceee2e..62a31af 100644
--- a/rgbdscheck.asm
+++ b/rgbdscheck.asm
@@ -1,8 +1,8 @@
-MAJOR EQU 0
-MINOR EQU 6
-PATCH EQU 0
+DEF MAJOR EQU 0
+DEF MINOR EQU 6
+DEF PATCH EQU 0
 
-WRONG_RGBDS EQUS "fail \"pokeyellow requires rgbds v0.6.0 or newer.\""
+DEF WRONG_RGBDS EQUS "fail \"pokeyellow requires rgbds v0.6.0 or newer.\""
 
 IF !DEF(__RGBDS_MAJOR__) || !DEF(__RGBDS_MINOR__) || !DEF(__RGBDS_PATCH__)
 	WRONG_RGBDS
-- 
2.52.0

